src/Microsoft.OData.Edm/Vocabularies/Annotations/EdmVocabularyAnnotation.cs

```diff
@@ -77,5 +289,13 @@ public IEdmExpression Value
         {
             get { return this.value; }
         }
+
+        /// <summary>
+        /// Gets whether the annotation uses a default value
+        /// (not defined with a provided value).
+        /// </summary>
+        public bool UsesDefault {

```

Comment:
>public bool UsesDefault { [](http://example.com/codeflow?start=8&length=25)

Since it's only "Getter". So, I don't think we need such interface.

Reply:
I think the main reason behind this was to allow us to support null values in the future.

Reply:
Yes. We can supply two set of constructor.

1) The constructor without "IEdmExpression", it means to use the default value from Term.

```C#
public EdmVocabularyAnnotation(IEdmVocabularyAnnotatable target, IEdmTerm term)
public EdmVocabularyAnnotation(IEdmVocabularyAnnotatable target, IEdmTerm term, string qualifier)
{
      useDefault = true;
}
```

2) The constructor with "IEdmExpression", it means to use the value from customer.
     We can also change the verification logic in these constructor to accept "null" value. The only difference is "useDefault".
     In the above, the useDefault = true
     in this case, the useDefault = false.     

```C#
public EdmVocabularyAnnotation(IEdmVocabularyAnnotatable target, IEdmTerm term, IEdmExpression value)
public EdmVocabularyAnnotation(IEdmVocabularyAnnotatable target, IEdmTerm term, string qualifier, IEdmExpression value)
{
      useDefault = false;
}
```

We can public "UseDefault" property, but, it seems there's no needed to add it into the interface, or create a new interface. That's my opinion.

Reply:
Ah, I think I see what you mean -- rather than add a getter method for UsesDefault (and therefore a new interface), simply make the property public.

Reply:
Adding a public property on EdmVocabularyAnnotation works for cases where *we* create the type representing the annotation (i.e., we could attempt to cast to EdmVocabularyAnnotation when we serialize, in order to read the UsesDefault property), but that doesn't allow other model implementations (CsdlVocabularyAnnotation or other custom implementations of IEdmVocabularyAnnottaion) to have a way to say that the value uses the default. The reason we define interfaces, along side the Edm* types, is so that we can support developers providing their own custom implementations.

In general, all of our deserialization should go through common, public interfaces, rather than rely on private implementation.

So, I would keep the new interface and implement it on EdmVocabularyAnnotation and CsdlVocabularyAnnotation, and check for it in the serializer.